pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/IngestionUtils.java

```diff
@@ -240,7 +240,7 @@ public static void uploadSegment(String tableNameWithType, BatchConfig batchConf
           }
           PinotFS outputFileFS = getOutputPinotFS(batchConfig, outputSegmentDirURI);
           Map<String, String> segmentUriToTarPathMap = SegmentPushUtils.getSegmentUriToTarPathMap(outputSegmentDirURI,
-              segmentUploadSpec.getPushJobSpec(), new String[] { segmentTarURIs.toString() });
+              segmentUploadSpec.getPushJobSpec(), segmentTarURIStrs.toArray(String[]::new));
```

Comment:
If you change this to new String[0] it will compile at JDK8 source level and the presto compatibility check will pass